lib/Tunnel.js

```diff
@@ -39,34 +39,32 @@ Tunnel.prototype._init = function(cb) {
     var assigned_domain = opt.subdomain;
 
     // where to quest
-    params.uri = base_uri + ((assigned_domain) ? assigned_domain : '?new');
+    uri = base_uri + ((assigned_domain) ? assigned_domain : '?new');
 
     (function get_url() {
-        request(params, function(err, res, body) {
-            if (err) {
-                // TODO (shtylman) don't print to stdout?
-                console.log('tunnel server offline: ' + err.message + ', retry 1s');
-                return setTimeout(get_url, 1000);
-            }
-
-            if (res.statusCode !== 200) {
-                var err = new Error((body && body.message) || 'localtunnel server returned an error, please try again');
+        axios.get(uri, params)
+        .then(function(res){
+            var body = res.data;
+            if (res.status !== 200) {
```

Comment:
What does axios consider a successful response vs an error? I believe there is a way to have axios consider all non-200 responses an error.

Reply:
fixed via bbdbfd6

Reply:
From [](https://github.com/axios/axios/blob/master/lib/adapters/http.js), axios considers various error conditions such as network timeout, improper request format, low-level network errors, etc. Everything else is technically considered by axios as a successful response.
In the [config](https://github.com/axios/axios#request-config) object, there's a `validateStatus` option that allows us to define conditions for accepting or rejecting the promise, but that wouldn't throw an error.
The code above currently calls the callback with the error for non-200 successful responses, and catches axios-defined errors using the catch() block further down.